packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -122,9 +122,10 @@ describe( 'Basic rendering', () => {
 			} );
 
 			// Click the "Edit" button to trigger into the editing mode.
-			const editButton = container.querySelector(
-				'.block-editor-link-control__search-item-action'
-			);
+			const editButton = Array.from(
+				container.querySelectorAll( 'button' )
+			).find( ( button ) => button.innerHTML.includes( 'Edit' ) );
```

Comment:
Searching by the element content seems to be a little bit fragile as well, but I guess we don't any other way to get it, such as `id`, a `data-` attribute, or whatever.

Reply:
I understand how this might appear. However, my take on this is that this is precisely when I'd _want_ my test to break.

If the text changes to `Flibble Wibble` then I want the test to say "Whoooooaaa hold on there Mr!, you changed a crucial piece of UI text". This would force me as a developer to acknowledge my change and update the test to reflect it (ie: it couldn't happen by accident).

Contrast this with any of the other approaches mentioned. Changing an `id` - user can't see it, makes no difference to them but the test fails. Test is coupled to an implementation detail.